pkg/ingester/ingester_v2.go

```diff
@@ -394,6 +431,51 @@ func (i *Ingester) v2MetricsForLabelMatchers(ctx old_ctx.Context, req *client.Me
 	return result, nil
 }
 
+func (i *Ingester) v2UserStats(ctx old_ctx.Context, req *client.UserStatsRequest) (*client.UserStatsResponse, error) {
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	db := i.getTSDB(userID)
+	if db == nil {
+		return &client.UserStatsResponse{}, nil
+	}
+
+	apiRate := db.ingestedAPISamples.rate()
+	ruleRate := db.ingestedRuleSamples.rate()
+	return &client.UserStatsResponse{
+		IngestionRate:     apiRate + ruleRate,
+		ApiIngestionRate:  apiRate,
+		RuleIngestionRate: ruleRate,
+		NumSeries:         db.Head().NumSeries(),
+	}, nil
```

Comment:
This code is duplicated with `v2AllUserStats()`. Would you mind adding a new function like `createUserStats(db)` which creates the user stat for 1 user, which we can directly `return` here and `append()` in the `v2AllUserStats()`, please?